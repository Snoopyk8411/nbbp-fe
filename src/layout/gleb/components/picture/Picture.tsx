import { GetServerSideProps } from 'next';
import Image from 'next/image';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPicture } from 'store/gleb/slice';
import { RootState } from 'store/reducers';
import store from 'store/index';
import imageLoader from 'layout/gleb/components/imageLoader';
import Layout from 'layout/gleb/components/layout/Layout';
import styles from './picture.module.css';

const Picture = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getPicture());
  }, []);

  const picture = useSelector((state: RootState) => state.picture);
  const date = picture.date.toString();

  const image = picture.url ? (
    <Image loader={imageLoader} unoptimized src={picture.url} alt={picture.title} width='500px' height='500px' />
  ) : (
    <img src={picture.url} alt={picture.title} className={styles.picture} />
  );

  return (
    <div>
      <div className={styles.main}>
        <h1 data-testid='title'>Astronomy picture of the day</h1>
        <div>{date}</div>
        {picture.media_type !== 'video' ? (
          image
        ) : (
          <iframe id='video' title='video' loading='lazy' width='500' height='500' src={picture.url}></iframe>
        )}
        <div className={styles.picture_title}>{picture.title}</div>
        <div>{picture.explanation}</div>
      </div>
    </div>
  );
};

Picture.getLayout = function getLayout(page: typeof Picture) {
  return <Layout>{page}</Layout>;
};

// export const getStaticProps = () => {
//   const picture = useSelector((state: RootState) => state.picture);
//   const initialState = picture;

//   return {
//     props: initialState,
//   };
// };

export default Picture;
