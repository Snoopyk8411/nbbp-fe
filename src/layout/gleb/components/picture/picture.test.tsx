import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Picture from 'layout/gleb/components/picture/Picture';

let mockStore: any;
const mockStoreConf = configureStore();

const mockData = {
  picture: {
    copyright: '',
    date: '2021-11-07',
    explanation: "To some it looks like a cat's eye.",
    hdurl: 'https://apod.nasa.gov/apod/image/2111/CatsEye_HubblePohl_1278.jpg',
    media_type: 'image',
    service_version: 'v1',
    title: "The Cat's Eye Nebula in Optical and X-ray",
    url: 'https://apod.nasa.gov/apod/image/2111/CatsEye_HubblePohl_960.jpg',
  },
};

describe('Picture Component', () => {
  beforeEach(() => {
    mockStore = mockStoreConf(mockData);
  });

  it('Picture snapshot', () => {
    const component = renderer.create(
      <Provider store={mockStore}>
        <Picture />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
