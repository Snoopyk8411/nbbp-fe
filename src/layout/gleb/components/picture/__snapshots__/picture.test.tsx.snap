// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Component Picture snapshot 1`] = `
<div>
  <div
    className="main"
  >
    <h1
      data-testid="title"
    >
      Astronomy picture of the day
    </h1>
    <div>
      2021-11-07
    </div>
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="The Cat's Eye Nebula in Optical and X-ray"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="The Cat's Eye Nebula in Optical and X-ray"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="https://apod.nasa.gov/apod/image/2111/CatsEye_HubblePohl_960.jpg"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <div
      className="picture_title"
    >
      The Cat's Eye Nebula in Optical and X-ray
    </div>
    <div>
      To some it looks like a cat's eye.
    </div>
  </div>
</div>
`;
